app/models/post.rb

```diff
@@ -80,7 +80,10 @@ class Post < ActiveRecord::Base
   register_custom_field_type(MISSING_UPLOADS_IGNORED, :boolean)
 
   scope :private_posts_for_user, ->(user) {
-    where("posts.topic_id IN (#{Topic::PRIVATE_MESSAGES_SQL})", user_id: user.id)
+    joins("LEFT JOIN topic_allowed_users ON topic_allowed_users.topic_id = posts.topic_id AND topic_allowed_users.user_id = #{user.id}")
```

Comment:
When interpolating strings for SQL it's always a good idea to add `.to_i` to avoid injection. I know in this case it would be unusual since it's coming from a `user` instance but who knows who will copy and paste or modify this SQL in the future.

Reply:
Too bad AR doesn't support the same interpolation method here as in `where("topic_id = ?", topic.id)`